src/state.cc

```diff
@@ -159,6 +159,23 @@ bool State::AddDefault(StringPiece path, string* err) {
   return true;
 }
 
+bool State::PathExistsBetween(Edge* from, Edge* to) {
```

Comment:
Better to detect cycle to prevent infinite recursion?

Reply:
Hm, is it possible to have cycles by this point? I thought we'd would error out before getting there. Needs checking in a test I guess.

Reply:
checked in a gtest that the code will error out before we get there